base.py

```diff
@@ -1064,14 +1064,32 @@ def update_gpu_test_expectations(expectation_file):
                         # otherwise comment the line and append the updated line.
                         # For the line already with 'intel' tag, keep as is if there is no duplicate line.
                         if updated_line in intel_lines:
-                            line = "# " + line
+                            line = '# ' + line
                         else:
                             if updated_gpu_tags != gpu_tags:
-                                line =  "# " + line + updated_line
+                                line =  '# ' + line + updated_line
                             intel_lines.append(updated_line)
             sys.stdout.write(line)
         fileinput.close()
 
+        # Locally maintained webgpu cts expectations
+        if not has_update_comment and expectation_file.endswith('webgpu-cts/expectations.txt'):
```

Comment:
We need to make it more general. For example, we can have a dict as below:
{'webgpu-cts/expectations.txt': ['xxx', 'xxy', 'xxz']}

We're going to maintain these local expectations from time to time. To put this in base.py will always result in 2 PR changes. Please put this in a standalone file in toolkit/misc/, and also move the function update_gpu_test_expectations() there.

Reply:
I made the changes in another PR: https://github.com/webatintel/toolkit/pull/34